app/test_main.py

```diff
@@ -2,0 +1,19 @@
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+def test_check_valid_url(
+        mocked_valid_google_url: mock.MagicMock
+) -> None:
+    can_access_google_page("https://www.youtube.com/")
+    mocked_valid_google_url.assert_called_once_with("https://www.youtube.com/")
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_check_internet_connection(
+        mocked_has_internet_connection: mock.MagicMock
+) -> None:
```

Comment:
The test name `test_check_internet_connection` could be more descriptive. Consider renaming it to something like `test_internet_connection_calls_has_internet_connection` to clearly indicate what aspect of the function is being tested.